parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java

```diff
@@ -218,6 +218,13 @@ private void writeAllFields(MessageOrBuilder pb) {
 
       for (Map.Entry<Descriptors.FieldDescriptor, Object> entry : changedPbFields.entrySet()) {
         Descriptors.FieldDescriptor fieldDescriptor = entry.getKey();
+
+        if(fieldDescriptor.isExtension()) {
+          // Field index of an extension field might overlap with a base field, thus
+          // they have to be ignored.
+          continue;
+        }
+
```

Comment:
Another option would be throw an Exception here and explicitly do not support messages.

I could also log a warning here, somehow only once though, to avoid the noise.